pyqt-apps/siriushla/widgets/timeplot.py

```diff
@@ -1,33 +1,89 @@
-import numpy as np
+import numpy as _np
 import time
+
 from qtpy.QtCore import Qt, Slot
 from qtpy.QtWidgets import QMenu
+
+from pyqtgraph import ViewBox
+
+from pydm import utilities
 from pydm.widgets.timeplot import TimePlotCurveItem, PyDMTimePlot, \
     DEFAULT_X_MIN
 
+from siriuspy.clientarch import ClientArchiver
+
 
 class SiriusTimePlotItem(TimePlotCurveItem):
     """Reimplement to do not receive inf values."""
 
+    def setBufferSize(self, value):
+        """Reimplement."""
+        if self._bufferSize == int(value):
+            return
+        self._bufferSize = max(int(value), 2)
+
+        old_data_buffer = _np.copy(self.data_buffer)
+        self.points_accumulated = old_data_buffer.shape[1]
+        self._min_y_value = min(old_data_buffer[1])
+        self._max_y_value = max(old_data_buffer[1])
+
+        self.data_buffer = _np.zeros((2, self._bufferSize),
```

Comment:
don't you think this style is uglier than:
```
self.data_buffer = _np.zeros(
    (2, self._bufferSize), order='f', dtype=float)
```

Reply:
yes, hahaha, my editor do this automatically, I will try to verify these cases